pkg/reconciler/taskrun/validate_resources.go

```diff
@@ -311,19 +311,27 @@ func validateTaskRunResults(tr *v1beta1.TaskRun, resolvedTaskSpec *v1beta1.TaskS
 }
```

Comment:
suggest a little more formatted version of the log message here:

```
		var s []string
		for k, v := range missmatchedTypes {
			s = append(s, fmt.Sprintf(" \"%v\": %v", k, v))
		}
		return fmt.Errorf("mismatched types: %v", strings.Join(s, ","))
```

This will return:

```
mismatched types:  "aResult": task result is expected to be "array" type but was initialized to a different type "string", "objectResult": task result is expected to be "object" type but was initialized to a different type "string"
```

Instead of:

```
missmatched Types for these results, map[aResult:want type: array, got type: string objectResult:want type: object, got type: string]
```

Reply:
yes! This is much better!